src/endpoints/withFetcher.js

```diff
@@ -20,30 +21,31 @@ function withFetcher (endpoint, hideError = false, hideSpinner = false) {
 
       constructor () {
         super()
-        this.state = { requesting: false }
+        this.state = { isDataAvailable: false }
       }
 
       fetch (options) {
         if (!options) {
           throw new Error('options are not valid! This could mean your mapStateToOptions() returns ' +
             'a undefined value!')
         }
-
-        this.setRequesting(true)
-        this.props.dispatch(endpoint.requestAction(options, options))
-          .then(() => { this.setRequesting(false) })
+        const isDataAvailable = this.props.dispatch(endpoint.requestAction(options, options))
+        this.setIsDataAvailable(isDataAvailable)
```

Comment:
setIsDataAvailable is called only once here. So if it is false here it will always stay false and the child component won't get displayed. So fetch() has to be called multiple times that the component state changes to true?

Reply:
Yep, the fetcher triggers the dispatch via fetch() which returns if the data is available yet. If not, the fetch-logic (in the endpoint) is triggered and once it's done the fetcher's componentWillUpdate will be triggered (and thus fetch() again) because the payload has been changed.

Reply:
do we need to trigger the dispatch after fetch() is called the second time?

Reply:
Well, it could (potentially) be, that the payload is still not correct. To assure, that no data is shown, until valid data is there, the dispatch is called a second time (where it only returns true if everything's valid).